ksqldb-parser/src/main/java/io/confluent/ksql/parser/tree/TableElements.java

```diff
@@ -91,6 +95,9 @@ public LogicalSchema toLogicalSchema() {
       if (tableElement.getConstraints().isKey() || tableElement.getConstraints().isPrimaryKey()) {
         builder.keyColumn(fieldName, fieldType);
       } else if (tableElement.getConstraints().isHeaders()) {
+        throwOnIncorrectHeaderColumnType(
```

Comment:
I think this check should be done earlier in the `AstBuilder` where the `TableElement` is created. My opinion is that once `TableElement` is created, then `toLogicalSchema` shouldn't fail. This method just converts from one object to another (unsure why isEmpty() fails, but it's already there). Also, the `AstBuilder` may check for invalid syntax, so why I think it should live there. What do you think?

Reply:
That makes sense. Moved.